docs/src/PlasmoSchwarz/algorithm.md

```diff
@@ -0,0 +1,42 @@
+
+Further details of the algorithm can be found in the original Plasmo.jl paper [here](https://arxiv.org/abs/2006.05378), but we will give an overview of the algorithm here. A graph $\mathcal{G}$ can be partitioned into $N$ subgraphs notated by $\{ \mathcal{SG}_i \}^N_{i=1}$. These subgraphs can be _expanded_(meaning that they will now include nodes within a prescribed distance of nodes in the original subgraph) to obtain _expanded subgraphs_ notated by $\{ \mathcal{SG}_i' \}^N_{i=1}$. Under Schwarz Decomposition, the constraints on the incident edges of the expanded subgraphs can be enforced using the previous iteration's solutions from the adjacent subgraphs. The formulation for a given subgraph, $\mathcal{SG}_i'$ is given by:
+
+```math
+\begin{align*}
+    \min&\;  \sum_{n \in \mathcal{N}(\mathcal{SG}_i')} f_n (x_n) - \sum_{e \in \mathcal{I}_1(\mathcal{SG}_i')} (\lambda_e^k)^\top g_e(\{x_n \}_{n \in \mathcal{N}(e) \cap\mathcal{N}(\mathcal{SG}_i')}, \{x_n^k \}_{n\in \mathcal{N}(e) \setminus \mathcal{N}(\mathcal{SG}_i')}) \\
+    \textrm{s.t.} &\; x_n \in \mathcal{X}_n, n \in \mathcal{N}(\mathcal{SG}_i') \\
+    &\; g_e(\{x_n \}_{n \in \mathcal{N}(e)}) = 0, e \in \mathcal{E}(\mathcal{SG}_i') \\
+    &\; g_e(\{x_n \}_{n \in \mathcal{N}(e) \cap\mathcal{N}(\mathcal{SG}_i')}, \{x_n^k \}_{n\in \mathcal{N}(e) \setminus \mathcal{N}(\mathcal{SG}_i')}) = 0, e \in \mathcal{I}_2(\mathcal{SG}_i')
+\end{align*}
+```
+
+Here, $(\cdot)^k$ denotes the iteration counter, $\lambda_e$ denotes the dual variable on their respective constraints, and $\mathcal{I}(\mathcal{SG}_i')$ denotes the set of incident edges (i.e., edges which contain nodes in the overlapped subgraph and outside of the overlapped subgraph). The sets $\mathcal{I}_1$ and $\mathcal{I}_2$ denote how the incident linking constraints are formulated in the subproblem (as either primal or dual coupled). 
+
+In the [thesis](https://asset.library.wisc.edu/1711.dl/V2UHW7KSFIKBQ8Q/R/file-e04b2.pdf) of Prof. Sungho Shin, it was also shown to improve convergence to include an augmented penalty term in the objective, and this is included in PlasmoSchwarz. The objective therefore also gets updated with the augmented term 
+
+```math
+\begin{align*}
+\sum_{e \in \mathcal{I}_1(\mathcal{SG}_i')} \mu \frac{1}{2} ||g_e(\{x_n \}_{n \in \mathcal{N}(e) \cap\mathcal{N}(\mathcal{SG}_i')}, \{x_n^k \}_{n\in \mathcal{N}(e) \setminus \mathcal{N}(\mathcal{SG}_i')})||
+\end{align*}
+```
+
+where $\mu$ is a parameter that can be set by the user. 
+
+Convergence of the algorithm is determined by primal and dual infeasibility which is computed at each iteration. The primal feasibility is determined by the incident linking constraints being feasible between the different subproblems, and the dual infeasibility is determined by the difference in the dual variables of the link constraints in the overlapping regions. 
```

Comment:
Also, @jalving can you confirm if this is correct? I believe this is how the dual and primal infeasiblities work.

Reply:
Correct. Primal feasibility evaluates the linking constraints between original (not expanded) subproblems. Dual feasibility evaluates consensus of the dual values for the same linking constraints for the expanded subproblems. Both checks use the infinity norm of the residuals.